torch/csrc/jit/ir.cpp

```diff
@@ -1454,11 +1454,15 @@ Node* Graph::createStringToFloat(Value* value) {
   return result;
 }
 
-Node* Graph::createClone(Node * n, std::function<Value*(Value*)> value_map, bool copy_blocks) {
+Node* Graph::createClone(
+    Node * n,
+    std::function<Value*(Value*)> value_map,
+    bool copy_blocks,
+    bool copy_unique_names) {
   //n can be from a different graph
   Node * r = n->allocNewInstance(this);
   for(auto o : n->outputs()) {
-    r->addOutput()->copyMetadata(o);
+    r->addOutput()->copyMetadata(o, copy_unique_names);
```

Comment:
Do we need this change to go this deep? Can't we simply have a method like `resetUniqueName()` that we call in the one place we actually need this?

Reply:
Good point. There are two callsites in Canonicalize that need this (createClone and copyMetadata) but I can use use resetUniqueName() there.

Reply:
Resolved